### PR TITLE
757: Support recording deployments in monitoring tools

### PR DESCRIPTION
**Pull request by stevejalim**
This changeset repurposes code from mdn/kuma to add rollout tracking to developerportal.

It will ping New Relic and Speedcurve when a deployment has taken place.

Note that because a deployment doesn't automatically trigger a rebuild of the static site (but this does happen automatically every 30 mins) it's possible/likely that speedcurve will profile the previous static build. This isn't a massive problem, but is something to think about. We could instead trigger Speedcurve from the static build pipeline

(Related issue #757)

## How to test

- the only real way to test this is to deploy it, so we should give it a REALLY close look-over before we do, and be prepared to tweak/fix when getting `master` onto staging once this is merged

### CODE REVIEW

**Comment by stevejalim** [comment]
@limed @escattone  I've made this as an early draft PR because am sure there will be subtleties that need fixing before this is merge ready. Please be as nitpicky as possible! ❤️

File: Jenkinsfile [hunk 1]
```diff
@@ -16,6 +16,19 @@ def notify_slack(Map args, credential_id='slack-hook-devportal-notifications') {
...
+    /*
+     * Record the rollout in external services like New Relic and SpeedCurve.
+     *
+     * Kept separate from deploy() so we don't double-log when deploying to
+     * an experimental EKS cluster as well as the regular k8s setup.
+     */
+
+      sh """
+        make k8s-record-deployment-job
+      """
```
Comment (stevejalim) [inline Jenkinsfile]: I _think_ this is right...

File: scripts/record_deployments.py [hunk 2]
```diff
@@ -0,0 +1,233 @@
+#!/usr/bin/env python
```
Comment (stevejalim) [inline scripts/record_deployments.py]: Do we have Python3 available?

File: scripts/record_deployments.py [hunk 3]
```diff
@@ -0,0 +1,233 @@
...
+from requests.auth import HTTPBasicAuth
+
+# API keys and IDs are set with environment variables rather than on the
+# command line, to avoid displaying the key in logs
+NR_API_KEY_NAME = "NEW_RELIC_API_KEY"
+NR_APP_IDS_NAME = "NEW_RELIC_APP_IDS"
+SC_API_KEY_NAME = "SPEEDCURVE_API_KEY"
+SC_SIDE_ID_NAME = "SPEEDCURVE_SITE_ID"
+
+APP_NAME = os.getenv("APP_NAME")  # "wagtail" is APP_NAME -- see Makefile's constants
```
Comment (stevejalim) [inline scripts/record_deployments.py]: Am not sure whether we should change "wagtail" to "developerportal" or not - does it matter massively?

File: scripts/record_deployments.py [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+import sys
+from argparse import ArgumentParser, RawDescriptionHelpFormatter
+from datetime import datetime
+
+import requests
+from requests.auth import HTTPBasicAuth
+
+# API keys and IDs are set with environment variables rather than on the
+# command line, to avoid displaying the key in logs
+NR_API_KEY_NAME = "NEW_RELIC_API_KEY"
```
Comment (stevejalim) [inline scripts/record_deployments.py]: @limed We are gonna need to get API keys (at least for Speedcurve) before we can merge this.

- [ ] @stevejalim Ticket up sourcing of API keys

File: k8s/Makefile [hunk 5]
```diff
@@ -139,6 +139,19 @@ k8s-jenkins-create-rbac:
 k8s-jenkins-delete-rbac:
 	kubectl delete -f jenkins-rbac.yaml
 
+k8s-record-deployment-job: k8s-delete-record-deployment-job
+	env APP_NAME=wagtail \
+		FROM_REVISION_HASH=${FROM_REVISION_HASH} \
+		TO_REVISION_HASH=$(shell make k8s-get-developerportal-revision-hash) \
+		j2 record-deployment-job.yaml.j2 | ${KC} apply -n ${K8S_NAMESPACE} -f -
+	env JOB_NAME=developerportal-record-deployment ./wait_for_job.sh
```
Comment (stevejalim) [inline k8s/Makefile]: Do we also need to record celery being booted at all?

File: scripts/record_deployments.py (hunk 2 above)
Comment (limed) [reply scripts/record_deployments.py]: We do on the jenkins server yeah

File: k8s/record-deployment-job.yaml.j2 [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+        - name: developerportal-{{ APP_NAME }}-record-deployment
+          image: {{ APP_IMAGE }}:{{ APP_IMAGE_TAG }}
+          imagePullPolicy: {{ APP_IMAGE_PULL_POLICY }}
+          resources:
+            requests:
+              cpu: 100m
+              memory: 256Mi
+            limits:
+              cpu: 500m
+              memory: 1Gi
```
Comment (limed) [inline k8s/record-deployment-job.yaml.j2]: I don't really think we need to give it that high of a limit, I am happy to say CPU `256m` and memory `512Mi`

File: scripts/record_deployments.py (hunk 2 above)
Comment (stevejalim) [reply scripts/record_deployments.py]: Is it the default `python` ?

**Comment by stevejalim** [comment]
@limed this can't be merged till we have API keys, of course

**Comment by stevejalim** [comment]
Blocking API key request ticketed as https://github.com/mdn/developer-portal/issues/879

File: scripts/record_deployments.py (hunk 2 above)
Comment (limed) [reply scripts/record_deployments.py]: Changed this to use `python3` in 4774d8550bf367e8bd868906e38e1f6c62aa7ccc

**Comment by stevejalim** [comment]
@limed Do you have any time to decide whether this is now ready to go, please? No rush, just housekeeping

**Comment by stevejalim** [comment]
@limed Heya dude - is this branch OK to merge now, please?

File: scripts/record_deployments.py (hunk 4 above)
Comment (stevejalim) [reply scripts/record_deployments.py]: @atopal Could you get hold of a Speedcurve API key off (off a service account?) please?

Comment (atopal) [reply scripts/record_deployments.py]: @stevejalim I thought we had that already, but I sent you the key via Slack

**Pull request closed without merging**